lib/Sema/DerivedConformanceCodable.cpp

```diff
@@ -89,6 +89,12 @@ static CodableConformanceType typeConformsToCodable(TypeChecker &tc,
   // Some generic types need to be introspected to get at their "true" Codable
   // conformance.
   auto canType = target->getCanonicalType();
+  if (auto referenceType = dyn_cast<ReferenceStorageType>(canType)) {
```

Comment:
@jrose-apple I'll look at `getAnyOptionalObjectType` — there are a lot of small things like that that I'm not aware of that can make the code shorter.

Also, `encodeIfPresent` just unwraps a layer of optionality from the type and encodes. If the value is `nil`, it won't encode it (as opposed to explicitly encoding `nil`); if it's non-`nil`, it will encode. This is pretty reasonable behavior for most encoding formats, and since the default is `decodeIfPresent` on decode as well, we've got symmetry there, and it's lenient by default.